### PR TITLE
fix: prevent overwriting of error from GetCallerIdentity when it's not nil

### PR DESCRIPTION
**Pull request by kbeniwal**
In Reconcile [here](https://github.com/nirmata/kyverno-aws-adapter/blob/8357d4135745656f326584eb662623ede15519bc/controllers/awsadapterconfig_controller.go#L111), error is overwritten when the error was not nil, [[doc reference](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/STS.html#getCallerIdentity-property:~:text=Object\)%20%E2%80%94-,the%20de%2Dserialized%20data%20returned%20from%20the%20request.%20Set%20to%20null%20if%20a%20request%20error%20occurs.%20The%20data%20object%20has%20the%20following%20properties%3A,-UserId%20%E2%80%94%20\(String)].

### CODE REVIEW

File: controllers/awsadapterconfig_controller.go [hunk 1]
```diff
@@ -108,7 +108,7 @@ func (r *AWSAdapterConfigReconciler) Reconcile(ctx context.Context, req ctrl.Req
 
 	callerIdentity, err := stsClient.GetCallerIdentity(ctx, &sts.GetCallerIdentityInput{})
 	if err != nil || callerIdentity == nil || callerIdentity.Account == nil {
-		if callerIdentity == nil || callerIdentity.Account == nil {
+		if err == nil {
 			err = fmt.Errorf("callerIdentity nil")
```
Comment (pns-nirmata) [inline controllers/awsadapterconfig_controller.go:112]: There should be two if statements. The existing one, and the new one printing error. I think there is a scenario when error is nil, but the callerIdentity and/or callerIdentity.Account is also nil. See the code change in: https://nirmata.slack.com/archives/D04PMMBUS0Z/p1676985624368149

Comment (kbeniwal) [reply controllers/awsadapterconfig_controller.go:116]: Apologies, I missed that case. Thanks for pointing it out!

File: controllers/awsadapterconfig_controller.go [hunk 2]
```diff
@@ -108,6 +108,10 @@ func (r *AWSAdapterConfigReconciler) Reconcile(ctx context.Context, req ctrl.Req
 
 	callerIdentity, err := stsClient.GetCallerIdentity(ctx, &sts.GetCallerIdentityInput{})
 	if err != nil || callerIdentity == nil || callerIdentity.Account == nil {
+		if err != nil {
+			l.Error(err, "GetCallerIdentity returned error")
+		}
+
```
Comment (pns-nirmata) [inline controllers/awsadapterconfig_controller.go:114]: else if needed here.

File: controllers/awsadapterconfig_controller.go [hunk 3]
```diff
@@ -108,6 +108,10 @@ func (r *AWSAdapterConfigReconciler) Reconcile(ctx context.Context, req ctrl.Req
 
 	callerIdentity, err := stsClient.GetCallerIdentity(ctx, &sts.GetCallerIdentityInput{})
 	if err != nil || callerIdentity == nil || callerIdentity.Account == nil {
+		if err != nil {
+			l.Error(err, "GetCallerIdentity returned error")
+		}
+		
 		if callerIdentity == nil || callerIdentity.Account == nil {
```
Comment (pns-nirmata) [inline controllers/awsadapterconfig_controller.go:115]: Can be changed to err == nil

Comment (kbeniwal) [reply controllers/awsadapterconfig_controller.go]: Done

**Review by pns-nirmata — APPROVED** [review]

**Pull request merged**